lib/Ceilometer/Types/Base.hs

```diff
@@ -0,0 +1,24 @@
+module Ceilometer.Types.Base where
+
+import           Control.Isomorphism.Partial.Iso
+import           Data.Bimap                      (Bimap)
+import qualified Data.Bimap                      as B
+import           Data.Word
+
+-- |Iso generator
+generateIsoFromBimap :: (Ord a, Ord b) => Bimap a b -> Iso a b
```

Comment:
From bijection to isomorphism, neat!

You probably want to use unsafeMakeIso' and stick error messages in there, otherwise if you stick this in a long chain of isos you will not know which one failed. That may require Show instances to make useful instances. If that's not okay, you can just  name it.